packages/bp/src/admin/ui/src/app/routes/SegmentHandler.tsx

```diff
@@ -0,0 +1,88 @@
+import segmentPlugin from '@analytics/segment'
+import Analytics, { AnalyticsInstance, PageData } from 'analytics'
+import hash from 'hash.js'
+import React from 'react'
+import { connect } from 'react-redux'
+import { UserState } from '~/user/reducer'
+
+const APP_NAME = 'ADMIN_ANALYTICS' // for reference, in case of second account
+const WRITE_KEY = 'zE0mct7hGOZRtCyImjX9vT1NJ2TpfyGF' // taken from Segment UI
+
+let analytics: AnalyticsInstance
+let loggedIn: boolean = false
+
+const extractUserHashFromUser = (user: UserState): string | undefined => {
+  if (user?.profile?.email) {
+    return hash
+      .sha256()
+      .update('botpressUserEmail' + user.profile.email)
+      .digest('hex')
+  }
+}
+
+const initSegmentAnalytics = async () => {
+  analytics = Analytics({
+    app: APP_NAME,
+    plugins: [
+      segmentPlugin({
+        writeKey: WRITE_KEY
+      })
+    ]
+  })
+
+  await analytics.page()
+
+  return analytics
+}
+
+const handleNewUserState = (user: UserState): void => {
+  const userEmailHash = extractUserHashFromUser(user)
+
+  if (!loggedIn && userEmailHash) {
+    loggedIn = true
+    // @ts-ignore
+    void analytics.identify(null, {
+      userEmailHash,
+      machineUUID: window.UUID
+    })
+  }
```

Comment:
If you really want to avoid awaiting something, please bubble up your call and add "void" or a tslint comment on the first item, not on each of them. I would suggest this:
```js
const handleNewUserState = (user: UserState): Promise<void> => {
  const userEmailHash = extractUserHashFromUser(user)
  if(loggedIn || !userEmailHash){
    return
  }

  loggedIn = true
  await analytics.identify(null, { userEmailHash, machineUUID: window.UUID })
}
```

Reply:
I think I originally misunderstood what you meant, so I did two changes for this : 

1.  I added a comment about the ts-lint ignore. I wasn't using it because of void, but because there's a typing issue with the library i'm using. 
2. I changed the organization to match what you wrote. I think it's clearer. 
Thanks!